src/blockdev.rs

```diff
@@ -61,17 +56,48 @@ fn get_partition_with_label(device: &str, label: &str) -> Result<Option<BlkDev>>
         eprint!("{}", String::from_utf8_lossy(&*result.stderr));
         bail!("lsblk of {} failed", device);
     }
-    let output: LsBlk =
-        serde_json::from_slice(&*result.stdout).chain_err(|| "decoding lsblk JSON")?;
+    let output = String::from_utf8(result.stdout).chain_err(|| "decoding lsblk output")?;
+
+    // walk each device in the output
+    let re = Regex::new("([A-Z-]+)=\"([^\"]+)\"").unwrap();
```

Comment:
You can use raw strings here if you prefer avoiding quote-escapes: `r#"a "raw" string"#`